src/HtmlProcessor/AbstractHtmlProcessor.php

```diff
@@ -60,16 +60,16 @@ abstract class AbstractHtmlProcessor
     protected $domDocument = null;
 
     /**
-     * @var \DOMXPath
+     * @var \DOMXPath|null
      */
-    protected $xPath = null;
+    private $xPath = null;
 
     /**
      * The constructor.
      *
      * Please use `::fromHtml` or `::fromDomDocument` instead.
      */
-    private function __construct()
+    final private function __construct()
```

Comment:
We can make it `protected`, but we cannot drop the `final` or drop the constructor altogether. If we do any of these, Psalm (rightly) complains:

```
ERROR: UnsafeInstantiation - src/HtmlProcessor/AbstractHtmlProcessor.php:106:21 - Cannot safely instantiate class Pelago\Emogrifier\HtmlProcessor\AbstractHtmlProcessor with "new static" as its constructor might change in child classes (see https://psalm.dev/229)
        $instance = new static();
```

Reply:
In that case I think we should use the `@psalm-consistent-constructor` annotation so that child classes can have a constructor if needed, but, if so, Psalm would enforce that it has the same parameters (i.e. no parameters).

I also think the constructor should be `protected` but that's probably beyond the scope of this PR.

Reply:
Okay, changed accordingly.